client/src/components/NavBar/NavBar.tsx

```diff
@@ -5,37 +5,57 @@ import logo from '../../Images/logo.png';
 import CircularProgress from '@material-ui/core/CircularProgress';
 import { useAuth } from '../../context/useAuthContext';
 import { useSocket } from '../../context/useSocketContext';
-import { useHistory } from 'react-router-dom';
-import { useEffect } from 'react';
+import { useHistory, useLocation } from 'react-router-dom';
+import { useEffect, useRef } from 'react';
 import LoggedInBar from './AuthBars/LoggedInBar';
 import LoggedOutBar from './AuthBars/LoggedOutBar';
 import CssBaseline from '@material-ui/core/CssBaseline';
+import HomePageBar from './AuthBars/HomePageBar';
 
 const NavBar = (): JSX.Element => {
   const classes = useStyles();
 
   const { loggedInUser } = useAuth();
   const { initSocket } = useSocket();
+  const outterNavStyle = useRef(classes.appbarHomePage);
+  const innerNavStyle = useRef(classes.appbarHomePage);
 
   const history = useHistory();
+  const location = useLocation();
 
   useEffect(() => {
     initSocket();
   }, [initSocket]);
 
+  useEffect(() => {
+    const path = history.location.pathname;
+    if (path === '/') {
+      outterNavStyle.current = classes.appbarHomePage;
```

Comment:
Optional: You can use a lightweight package like [clsx](https://www.npmjs.com/package/clsx) to conditionally apply styles to components, I think it would make the code a bit cleaner :)

Reply:
Really interesting, thank you - I'm going to try and implement this one.  Yeah I was struggling to figure out the clearnest way to implement that conditional styling.